app/src/main/java/com/example/recipeapp/ReviewsActivity.java

```diff
@@ -0,0 +1,71 @@
+package com.example.recipeapp;
+
+import androidx.activity.result.ActivityResult;
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.FileProvider;
+import androidx.lifecycle.Observer;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+
+import com.example.recipeapp.Models.Recipe;
+import com.example.recipeapp.databinding.ActivityReviewsBinding;
+import com.example.recipeapp.viewmodels.ReviewsViewModel;
+
+import java.io.File;
+
+public class ReviewsActivity extends AppCompatActivity {
+    private static final String TAG = "ReviewsActivity";
+    ActivityReviewsBinding binding;
+    ReviewsViewModel viewModel;
+    ActivityResultLauncher<Intent> galleryLauncher;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = ActivityReviewsBinding.inflate(getLayoutInflater());
+        String title = getIntent().getStringExtra(DetailsActivity.RECIPE);
+        Long id = getIntent().getLongExtra(DetailsActivity.RECIPE_ID, 0);
+        binding.setViewModel(new ReviewsViewModel(title, getString(R.string.write_review), id));
+        binding.executePendingBindings();
```

Comment:
noob question, what does this api do ?

Reply:
https://stackoverflow.com/questions/53043412/android-why-use-executependingbindings-in-recyclerview
Forces to load all changes in view